### DISCUSSION TITLE
main.py returns error 13 permission denied when running on ubuntu 22.04

### DISCUSSION THREAD

**Discussion by abhandari73** [zauberzeug/nicegui#3043] [Q&A]
### Question

I have a web server running on my local machine - apache2 on ubuntu and i can see the static pages and php pages from outside the network. I created a basic script to be rendered on the webpage but when i run python3 main.py from the directory in which the script is saved i get Error 13 Permission denied. I changed the permissions to 775. I am running with admin privileges.

The code is as below:
```py
from nicegui import ui
import numpy as np
from matplotlib import pyplot as plt

def main() -> None:

        ui.icon('settings')
        with ui.pyplot(figsize=(3,2)):
                x = np.linspace(0,5)
                y = np.cos(2*np.pi*x)*np.exp(-x)
                plt.plot(x,y, '-')
        
        ui.notify('my notification')

if __name__ in ('__main__', '__mp_main__'):
        main()
        ui.run(on_air=True, port=80)
```

What am I doing wrong and why is it not being rendered on the apache server

**Comment by rodja** [comment]
By default you are not allowed to start services on port 80 as a normal user. You could use tools like setcap, authbind or iptables. An other alternative would be a reverse proxy running as root which serves your app which then can run on a different port. We often use a third way via docker. This works because docker runs as root and we can bind the internal port (say 8080) to the outside port 80.
